hack/update-mocks.sh

```diff
@@ -139,4 +139,8 @@ git ls-files -mo --exclude-standard -z | while read -r -d $'\0' file; do
     cat hack/boilerplate/boilerplate.generatego.txt "$file" >"${tmp}" && \
     mv "${tmp}" "$file"
   fi
+
+  # fix up the copyright template - it ends up with the value of temp_file_name,
+  # replace that with the $BUILD_TAG_FILE placeholder used in the source files
+  sed -i -E "s/-copyright_file=[^ ]*$(basename "$0")\.[^ ]{6}/-copyright_file=\$BUILD_TAG_FILE/" "$file"
```

Comment:
I'm trying to understand what changed to make this necessary...

Reply:
do we know if something changed / broke in https://github.com/golang/mock between 1.6.0 and the last change to mockgen (https://github.com/golang/mock/commits/main/mockgen)

or if something changed / broke in https://github.com/uber-go/mock/commits/main/mockgen since it picked up dev from 5b455625bd2c8ffbcc0de6a0873f864ba3820904?

Reply:
`mockgen` now keeps a trace of the command used to generate each generated file, which in our case includes the temporary file name used to build the copyright/license header; see https://github.com/kubernetes/kubernetes/pull/120969/files#diff-2a971cd6e80c8c54aa8c0f04dfbf50e5c8a48a070c1cae0c86c525bb9c3e7427R22 for example.

Reply:
This changes was introduced by https://github.com/uber-go/mock/pull/31 which fixed https://github.com/golang/mock/issues/554.

Reply:
If we are using sed, please ensure we are using https://github.com/kubernetes/kubernetes/blob/f19b62fc0914b38941922afefd1e34eb55f87ee7/hack/lib/util.sh#L761-L774 so that it's bsd/macos compatible. Thanks!

Reply:
looks like there was an attempt made to make the recorded invocation reproducible / stable ... I asked in https://github.com/uber-go/mock/issues/80#issuecomment-1749197366 if it could be opted out of

let's at least see if there's movement on that side before going this `sed` route

Reply:
Right, post-processing generated code should be avoided. Perhaps if this ultimately has to be dealt with in `k/k`, a better solution would be to ignore the comments in the validation comparison. (But skipping the record of the command used to generate the code would be better.)